sdk/core/azure-core/inc/azure/core/internal/json/json_serializable.hpp

```diff
@@ -25,6 +25,14 @@ namespace Azure { namespace Core { namespace Json { namespace _internal {
      * @return The json string representation.
      */
     virtual std::string Serialize() const = 0;
+
+    /**
+     * @brief Default destructor for serializable objects.
+     *
+     * @remark This makes the type compatible with Response<T>
+     *
```

Comment:
Please help me understand why we need this.

Reply:
Imagine we create a derived class `D` from this class.
Then, at some point, we cast `D` to `Serializable` (like to pass it to one function)
```cpp
class D : public Serializable {
};

void Foo(Serializable value) {
}

// Call Foo
D instance;
Foo(instance);
```

Then, when `instance` behaves like Serializable, and gets destroyed, the compiler needs to know what destructor to call. If there is not a dttor in the base class, the compiler would give an error saying `I don't know how to destroy the object`

Reply:
This `@remark` is confusing me: `This makes the type compatible with Response<T>`
What do you mean by that?

Reply:
After the last refactoring to move all serializers/deserialisers to static methods, I think I don't need this class anymore

Reply:
at least the dttor is not required anymore.  I will remove the internal class for preview 2